services/personal/api.go

```diff
@@ -1,91 +1,91 @@
 package personal
 
 import (
-	"context"
 	"errors"
-	"strings"
-	"time"
 
-	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
-
 	"github.com/status-im/status-go/account"
+	"github.com/status-im/status-go/eth-node/crypto"
 	"github.com/status-im/status-go/eth-node/types"
-	"github.com/status-im/status-go/params"
-	"github.com/status-im/status-go/rpc"
 )
 
 var (
-	// ErrInvalidPersonalSignAccount is returned when the account passed to
-	// personal_sign isn't equal to the currently selected account.
-	ErrInvalidPersonalSignAccount = errors.New("invalid account as only the selected one can generate a signature")
+	ErrInvalidSignatureLength = errors.New("invalid signature, must be 65 bytes long")
+	ErrInvalidSignatureV      = errors.New("invalid Ethereum signature (V is not 27 or 28)")
 )
 
-// SignParams required to sign messages
-type SignParams struct {
-	Data     interface{} `json:"data"`
-	Address  string      `json:"account"`
-	Password string      `json:"password"`
-}
-
-// RecoverParams are for calling `personal_ecRecover`
-type RecoverParams struct {
-	Message   string `json:"message"`
-	Signature string `json:"signature"`
+// This interface is used to mock the personal API for testing purposes
+// without it in many places in tests we would need to create verifiedAccount (including a proper account manager setting) and
+// that will also require pdating all fake addresses/signatures etc. we used in communities/messenger tests.
+type PersonalAPIInterface interface {
+	Recover(rpcParams RecoverParams) (addr types.Address, err error)
+	CanRecover(rpcParams RecoverParams, revealedAddress types.Address) (bool, error)
+	Sign(rpcParams SignParams, verifiedAccount *account.SelectedExtKey) (result types.HexBytes, err error)
```

Comment:
I think this interface should be defined in `Messenger` instead. That would be Go-idiomatic way.
Because that's the requirement of `Messenger`: provide something that implements such interface.

Reply:
Correct. Moving this there.

Reply:
@igor-sirotin changes applied.